### PR TITLE
Added option for setting video resolution, moved webcamDim option to …

### PR DESCRIPTION
**Pull request by exaroth**
This adds videoRes option for setting video resolution in dialog, also moved webcamDim option to a 'webcam' group.

### CODE REVIEW

File: src/widgets/widgets.js [hunk 1]
```diff
@@ -86,15 +89,19 @@ filepicker.extend('widgets', function(){
...
 
-        setAttrIfExistsArray(fpoptions, domElement, generalOptionsMap);
+        setAttrIfExistsArray(fpoptions, domElement, generalOptionsMap, null);
 
         if (mode === 'export') {
             setAttrIfExists('suggestedFilename', fpoptions, 'data-fp-suggestedFilename', domElement);
         } else if (mode === 'pick') {
-            setAttrIfExistsArray(fpoptions, domElement, pickOnlyOptionsMap);
+            setAttrIfExistsArray(fpoptions, domElement, pickOnlyOptionsMap, null);
+            setAttrIfExistsArray(fpoptions, domElement, generalOptionsMap, "webcam")
```
Comment (krystiangw) [inline src/widgets/widgets.js]: Use rather:
```
fpoptions.webcamOptions = {};
setAttrIfExistsArray(fpoptions.webcamOptions, domElement, webcamOptionsMap);
```
So you won't have to change ```setAttrIfExistsArray``` function.

File: src/widgets/widgets.js [hunk 2]
```diff
@@ -69,7 +72,7 @@ filepicker.extend('widgets', function(){
             'data-fp-show-close': 'showClose',
             'data-fp-conversions': 'conversions',
             'data-fp-custom-text': 'customText',
-            'data-fp-custom-source-conatiner': 'customSourceContainer',
+            'data-fp-custom-source-container': 'customSourceContainer',
```
Comment (krystiangw) [inline src/widgets/widgets.js]: Good catch!

File: src/library/urls.js [hunk 3]
```diff
@@ -31,7 +31,8 @@ filepicker.extend('urls', function(){
             (options.storePath ? '&storePath='+options.storePath : '')+
             (options.storeContainer ? '&storeContainer='+options.storeContainer : '')+
             (options.storeAccess ? '&storeAccess='+options.storeAccess : '')+
-            (options.webcamDim ? '&wdim='+options.webcamDim.join(',') : '')+
+            ((options.webcam && options.webcam.webcamDim) ? '&wdim='+options.webcam.webcamDim.join(',') : '')+
```
Comment (krystiangw) [inline src/library/urls.js]: What about backward compatibility?

Comment (exaroth) [inline src/library/urls.js]: Slawek said it's okay

**Comment by szwacz** [comment]
:+1:

File: src/widgets/widgets.js (hunk 1 above)
Comment (exaroth) [inline src/widgets/widgets.js]: Fixed

**Comment by krystiangw** [comment]
:+1:

File: src/library/urls.js [hunk 4]
```diff
@@ -18,6 +18,7 @@ filepicker.extend('urls', function(){
     var allowedConversions = ['crop', 'rotate', 'filter'];
 
     var constructPickUrl = function(options, id, multiple) {
+        console.log(options);
```
Comment (krystiangw) [inline src/library/urls.js]: ```console.log``` left.

**Pull request merged**